lib/writeUris.js

```diff
@@ -0,0 +1,126 @@
+    for (var id in images) {
+        if (images.hasOwnProperty(id)) {
+            texturesByteLength += images[id].extras._obj2gltf.source.length;
+        }
+    }
+
+    // Buffers larger than ~192MB cannot be base64 encoded due to a NodeJS limitation. Source: https://github.com/nodejs/node/issues/4266
+    var exceedsMaximum = (texturesByteLength + bufferByteLength > 201326580);
+
+    if (exceedsMaximum) {
+        logger('Buffers and textures are too large to encode in the glTF, saving as separate resources.');
+    }
+
+    if (separateBuffers || exceedsMaximum) {
+        promises.push(writeSeparateBuffer(gltf, gltfPath));
+    } else {
+        writeEmbeddedBuffer(gltf);
+    }
+
+    if (separateTextures || exceedsMaximum) {
+        promises.push(writeSeparateTextures(gltf, gltfPath));
+    } else {
+        writeEmbeddedTextures(gltf);
+    }
+
+    return Promise.all(promises)
+        .then(function() {
+            deleteExtras(gltf);
+            return gltf;
+        });
+}
+
+function deleteExtras(gltf) {
+    var buffer = gltf.buffers[Object.keys(gltf.buffers)[0]];
+    delete buffer.extras;
+
+    var images = gltf.images;
+    for (var id in images) {
+        if (images.hasOwnProperty(id)) {
+            var image = images[id];
+            delete image.extras;
+        }
+    }
+}
+
+function writeSeparateBuffer(gltf, gltfPath) {
+    var buffer = gltf.buffers[Object.keys(gltf.buffers)[0]];
+    var source = buffer.extras._obj2gltf.source;
+    var bufferName = path.basename(gltfPath, path.extname(gltfPath));
+    var bufferUri = bufferName + '.bin';
+    buffer.uri = bufferUri;
+    var bufferPath = path.join(path.dirname(gltfPath), bufferUri);
+    return writeUris._outputFile(bufferPath, source);
+}
+
+function writeSeparateTextures(gltf, gltfPath) {
+    var promises = [];
+    var images = gltf.images;
+    for (var id in images) {
+        if (images.hasOwnProperty(id)) {
+            var image = images[id];
+            var extras = image.extras._obj2gltf;
+            var imageUri = image.name + extras.extension;
+            image.uri = imageUri;
+            var imagePath = path.join(path.dirname(gltfPath), imageUri);
+            promises.push(writeUris._outputFile(imagePath, extras.source));
+        }
+    }
+    return Promise.all(promises);
+}
+
+function writeEmbeddedBuffer(gltf) {
+    var buffer = gltf.buffers[Object.keys(gltf.buffers)[0]];
+    var source = buffer.extras._obj2gltf.source;
+    buffer.uri = 'data:application/octet-stream;base64,' + source.toString('base64');
+}
+
+function writeEmbeddedTextures(gltf) {
+    var promises = [];
+    var images = gltf.images;
+    for (var id in images) {
+        if (images.hasOwnProperty(id)) {
+            var image = images[id];
+            var extras = image.extras._obj2gltf;
+            image.uri = 'data:' + mime.lookup(extras.extension) + ';base64,' + extras.source.toString('base64');
+        }
+    }
+    return Promise.all(promises);
```

Comment:
Copy/paste error.  This function is synchronous and there's no need to create or return a Promise.  The promises array is never used.